server/src/main/java/org/opensearch/plugins/SecureTransportSettingsProvider.java

```diff
@@ -52,6 +54,18 @@ default Optional<SecureTransportParameters> parameters(Settings settings) {
     @ExperimentalApi
     interface SecureTransportParameters {
         boolean dualModeEnabled();
+
+        KeyManagerFactory keyManagerFactory();
+
+        String sslProvider();
+
+        String clientAuth();
+
+        Iterable<String> protocols();
```

Comment:
What is the purpose of `Iterable` here? May just simple array is does fit?
```suggestion
        String[] protocols();
```